### PR TITLE
Fetching notes from DB and added functionality to navbar

### PR DESCRIPTION
**Pull request by prachi-sahani**
This PR includes the following features:
 - sidenav - can be viewed by toggling menu button in navbar
 - user can now view notes in two views - grid view and list view
 - notes are now being fetched from backed
 - notes page can only be accessed if the user is logged in

Preview link - https://cakeopedia-6q656p4m0-prachi-sahani.vercel.app/

### CODE REVIEW

**Review by FarhanMobashir — APPROVED** [review]
Well structured and consistent code throughout.
Can be merged

File: src/components/ViewNotes.jsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+          </div>
+          <div className="card-footer note-footer">
+            <div className="action-icons">
+              <button className="btn-icon material-icons ">color_lens</button>
+              <button className="btn-icon material-icons ">content_copy</button>
+              <button className="btn-icon material-icons ">archive</button>
+              <button className="btn-icon material-icons ">delete</button>
+            </div>
+          </div>
+        </div> */}
```
Comment (Hamdrive) [inline src/components/ViewNotes.jsx]: Please clean up commented code

File: src/context/authorization-context.jsx [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+  const [authToken, setAuthToken] = useState("");
   const location = useLocation();
   const navigate = useNavigate();
   const [isLoadingLoginAsGuest, setIsLoadingLoginAsGuest] = useState(false);
   const [isLoadingLoginUser, setIsLoadingLoginUser] = useState(false);
   const [isLoadingSignup, setIsLoadingSignup] = useState(false);
   const { showSnackbar } = useMessageHandling();
+  useEffect(() => {
+    setAuthToken(sessionStorage.getItem("uid") || "");
+  }, []);
```
Comment (Hamdrive) [inline src/context/authorization-context.jsx]: Consider keeping this initialization within the state itself, using nullish coalescene

File: src/context/message-handling-context.jsx [hunk 3]
```diff
@@ -5,6 +5,7 @@ const MessageHandlingContext = createContext();
 function MessageHandlingProvider({ children }) {
   const [errorMessage, setErrorMessage] = useState("");
   const [showSidenav, setShowSidenav] = useState(false);
+  const [gridView, setGridView] = useState(true);
```
Comment (Hamdrive) [inline src/context/message-handling-context.jsx:8]: consider either having a separate context or better if prop drilling seems feasible upto a couple of levels then that would be better

File: src/pages/Notes.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+export function Notes() {
+  return (
+    <div>
+      <Navbar />
+      <div className="page-content">
+        <SideNav />
+        <main className="main-content">
+          <Routes>
+            <Route path="/" element={<ViewNotes />}></Route>
+            <Route path="*" exact={true} element={<PageNotFound />} />
```
Comment (Hamdrive) [inline src/pages/Notes.jsx:15]: exact does not make sense for wildcard route, can instead be used for notes subroutes (based on tags etc)

**Review by Hamdrive — COMMENTED** [review]
Overall neat code written
Good use of routes, context and component structure
Have suggested some improvements to code

File: src/components/ViewNotes.jsx (hunk 1 above)
Comment (prachi-sahani) [reply src/components/ViewNotes.jsx]: Done

File: src/context/authorization-context.jsx (hunk 2 above)
Comment (prachi-sahani) [reply src/context/authorization-context.jsx]: done

**Pull request merged**